flocker/node/agents/test/test_blockdevice.py

```diff
@@ -1264,6 +1273,483 @@ def test_all_transitions(self, desired_state, discovered_state):
+            block_device_api=self.api,
+            mountroot=mountroot_for_test(self),
+        )
+
+    def test_no_manifestations(self):
+        """
+        If there are no Manifestations on this node, then
+        there are no desired datasets calculated.
+        """
+        assert_desired_datasets(
+            self, self.deployer,
+            desired_manifestations=[],
+            expected_datasets=[],
+        )
+
+    def test_manifestation(self):
+        """
+        If there is a manifesation on this node, then the
+        corresponding dataset has a desired state of ``MOUNTED``.
+        """
+        assert_desired_datasets(
+            self, self.deployer,
+            desired_manifestations=[ScenarioMixin.MANIFESTATION.transform(
+                ['dataset', 'metadata'], ScenarioMixin.METADATA,
+            )],
+            expected_datasets=[
+                DesiredDataset(
+                    state=DatasetStates.MOUNTED,
+                    dataset_id=ScenarioMixin.DATASET_ID,
+                    metadata=ScenarioMixin.METADATA,
+                    maximum_size=int(REALISTIC_BLOCKDEVICE_SIZE.bytes),
+                    mount_point=self.deployer.mountroot.child(
+                        unicode(ScenarioMixin.DATASET_ID)
+                    ),
+                ),
+            ],
+        )
+
+    def test_manifestation_default_size(self):
+        """
+        If there is a manifesation on this node, then the
+        corresponding dataset has a desired state of ``MOUNTED``.
+        """
+        expected_size = int(GiB(75).bytes)
+        assert_desired_datasets(
+            self, self.deployer,
+            desired_manifestations=[
+                ScenarioMixin.MANIFESTATION.transform(
+                    ["dataset", "maximum_size"], lambda _: None,
+                    ['dataset', 'metadata'], ScenarioMixin.METADATA,
+                ),
+            ],
+            expected_datasets=[
+                DesiredDataset(
+                    state=DatasetStates.MOUNTED,
+                    dataset_id=ScenarioMixin.DATASET_ID,
+                    metadata=ScenarioMixin.METADATA,
+                    maximum_size=expected_size,
+                    mount_point=self.deployer.mountroot.child(
+                        unicode(ScenarioMixin.DATASET_ID)
+                    ),
+                ),
+            ],
+        )
+
+    def test_deleted_manifestation(self):
+        """
+        If there is a mounted dataset that is deleted on this
+        node, the corresponding dataset has a desired state of
+        ``DELETED``.
+        """
+        assert_desired_datasets(
+            self, self.deployer,
+            desired_manifestations=[
+                ScenarioMixin.MANIFESTATION.transform(
+                    ["dataset", "deleted"], True,
+                    ['dataset', 'metadata'], ScenarioMixin.METADATA,
+                ),
+            ],
+            # XXX FLOC-1772 We only need to do this to handle
```

Comment:
FLOC-1772 has been fixed, I believe, so can just delete this comment? Or reduce number of tests?